crates/turbopath/src/relative_unix_path_buf.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    ffi::OsStr,
+    path::{Components, Path, PathBuf},
+};
+
+use serde::Serialize;
+
+use crate::{IntoUnix, PathValidationError};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Default, Serialize)]
+pub struct RelativeUnixPathBuf(PathBuf);
+
+impl RelativeUnixPathBuf {
+    /// Create a new RelativeUnixPathBuf from a PathBuf by calling `into_unix()`
+    ///
+    /// NOTE: `into_unix` *only* converts Windows paths to Unix paths *on* a
+    /// Windows system. Do not pass a Windows path on a Unix system and
+    /// assume it'll be converted.
+    ///
+    /// # Arguments
+    ///
+    /// * `path`:
+    ///
+    /// returns: Result<RelativeUnixPathBuf, PathValidationError>
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// ```
+    pub fn new(path: impl Into<PathBuf>) -> Result<Self, PathValidationError> {
```

Comment:
Should we only be compiling this for Windows to avoid misuse?

Reply:
As above, we expect this to be called on unix systems, but to effectively be a no-op.